staging/src/k8s.io/kubectl/pkg/cmd/events/events.go

```diff
@@ -259,36 +323,22 @@ func (o EventsOptions) runWatch(namespace string, listOptions metav1.ListOptions
 	return nil
 }
 
-func printHeadings(w io.Writer, allNamespaces bool) {
-	if allNamespaces {
-		fmt.Fprintf(w, "NAMESPACE\t")
+// filteredEventType checks given event can be printed
+// by comparing it in filtered event flag.
+// If --event flag is not set by user, this function allows
+// all events to be printed.
+func (o *EventsOptions) filteredEventType(et string) bool {
+	if len(o.FilterTypes) == 0 {
+		return true
 	}
-	fmt.Fprintf(w, "LAST SEEN\tTYPE\tREASON\tOBJECT\tMESSAGE\n")
-}
 
-func printOneEvent(w io.Writer, e corev1.Event, allNamespaces bool) {
-	var interval string
-	firstTimestampSince := translateMicroTimestampSince(e.EventTime)
-	if e.EventTime.IsZero() {
-		firstTimestampSince = translateTimestampSince(e.FirstTimestamp)
-	}
-	if e.Series != nil {
-		interval = fmt.Sprintf("%s (x%d over %s)", translateMicroTimestampSince(e.Series.LastObservedTime), e.Series.Count, firstTimestampSince)
-	} else if e.Count > 1 {
-		interval = fmt.Sprintf("%s (x%d over %s)", translateTimestampSince(e.LastTimestamp), e.Count, firstTimestampSince)
-	} else {
-		interval = firstTimestampSince
-	}
-	if allNamespaces {
-		fmt.Fprintf(w, "%v\t", e.Namespace)
+	for _, t := range sets.NewString(o.FilterTypes...).List() {
```

Comment:
I'd suggest de-duplicating the list already in `Complete` and then also check the values in `Validate` so that here you'll be working only with a correct list of values.

Reply:
Good idea